### PR TITLE
Add UnionPay 8 series bins (Fixes #95)

### LINKED ISSUE #95: Missing 8 series bin range for UnionPay
### General information

* Library version: 8.2.0
* Browser and OS: Any

### Issue description

Announcement from the [Vantiv processor website](http://info.vantiv.com/rs/048-BUR-972/images/July%202018%20Enterprise%20Advantage%20Bulletin.html):

> China Union Pay is launching a new 8 Series BIN range. Merchants and partners should ensure all point-of-sale devices are ready to identify, accept, and process these new BINs by October 12, 2018. The 8 Series Cards are expected to begin appearing in the market January 2019.
>
>Start|End|Issuing Network
> -- | -- | --
> 81000000 | 81099999 | UnionPay
> 81100000 | 81319999 | UnionPay
> 81320000 | 81519999 | UnionPay
> 81520000 | 81639999 | UnionPay
> 81640000 | 81719999 | UnionPay

Looks like there is support for 6 series, but 8 has not been added yet.

### PR DESCRIPTION
**Pull request by leebradley**
Added UnionPay 8 bin series based on ranges given on Vantiv's website.

### CODE REVIEW

File: test/index.js [hunk 1]
```diff
@@ -138,6 +138,11 @@ describe('creditCardType', function () {
       ['6221558812340000', 'unionpay'],
       ['6269992058134322', 'unionpay'],
       ['622018111111111111', 'unionpay'],
+      ['8100513433325374', 'unionpay'],
+      ['8111700872004845', 'unionpay'],
+      ['8141618644273338', 'unionpay'],
+      ['8158163233706018', 'unionpay'],
+      ['8168524506870054', 'unionpay'],
```
Comment (crookedneighbor) [inline test/index.js]: Can we add a test for `8` being unionpay? Pretty sure it's the only one that can start with `8`.

File: lib/card-types.js [hunk 2]
```diff
@@ -115,7 +115,12 @@ var cardTypes = {
       [627781, 627799],
       [6282, 6289],
       6291,
-      6292
+      6292,
+      810,
```
Comment (crookedneighbor) [inline lib/card-types.js]: I think w should be able to do

```
8,
81,
```

as well since it's the only card that starts with 8.

Comment (crookedneighbor) [reply lib/card-types.js]: Wait, I think we adjusted the parsing to allow 810 to just work for 8. I'll have to check.

File: test/index.js (hunk 1 above)
Comment (leebradley) [reply test/index.js]: Done! Looks like you're right with your second comment about the parsing below. It was correctly identified as UnionPay.

**Comment by leebradley** [comment]
@crookedneighbor Are there any other changes that need made?

**Comment by crookedneighbor** [comment]
Not on your end.

Braintree doesn't support series 8 union pay cards, so we just need to prep our SDKs that use this module to validate it correctly. I hope to get this merged in and released next week.

**Comment by crookedneighbor** [comment]
This is out in v8.3.0. Thanks!

**Comment by richardags** [comment]
Where you guys find the BIN of all cards?

I only can find for Discovery from vendor at link above.

DISCOVER/UPDATED at March 2019 https://www.discoverglobalnetwork.com/downloads/IPP_VAR_Compliance.pdf

I find out others websites with different info of bin cards, how to compare what it's used actually?
https://pt.stackoverflow.com/questions/97234/validar-n%C3%BAmero-cart%C3%A3o-de-cr%C3%A9dito
https://howtodoinjava.com/regex/java-regex-validate-credit-card-numbers/

**Comment by leebradley** [comment]
@richardags Processors make announcements. At least that was the case with UnionPay. There's a link and a table in the referenced issue.

**Pull request merged**